jonprl-mode.el

```diff
@@ -48,6 +48,28 @@
   :type 'file
   :group 'jonprl)
 
+(defcustom jonprl-pretty-symbols
+  '(("*" . ?×)
+   ("->" . ?→)
+   (":" . ?∶)
+   ("<" . ?⟨)
+   ("<>" . ?⋄)
+   ("=>" . ?⇒)
+   ("=def=" . ?≝)
+   (">" . ?⟩)
+   ("[" . ?⸤)
+   ("]" . ?⸥)
```

Comment:
I think how the replacement works may (at least for `prettify-symbols-mode`) be partially controlled by the syntax table since some combination of symbols adjacent to a delimiter still triggers the replacement. It may be possible to get this working somehow but I'm not sure. In any case, the delimiter replacements aren't as important to be to be honest.